schema.cc

```diff
@@ -254,7 +254,7 @@ schema::raw_schema::raw_schema(utils::UUID id)
     : _id(id)
 { }
 
-schema::schema(const raw_schema& raw, std::optional<raw_view_info> raw_view_info)
+schema::schema(const raw_schema& raw, std::optional<raw_view_info> raw_view_info, bool is_index, bool is_global_index)
```

Comment:
Did we really need to add these as new separate parameters to the constructor? Wouldn't it be simpler if they were inside raw_view_info? The question of whether something is an index or not is a sub-question of whether it is a view (it first has to be a view - and if it's a view, it might be an index).
If this was inside raw_view_info, I think many of the other changes below will not be needed or simplified?